plugins/ltac/extratactics.mlg

```diff
@@ -1123,3 +1123,31 @@ let tclOPTIMIZE_HEAP =
 TACTIC EXTEND optimize_heap
 | [ "optimize_heap" ] -> { tclOPTIMIZE_HEAP }
 END
+
+(** tactic analogous to "Strategy" *)
+
+{
+
+(* Like Vernacentries.vernac_set_strategy, but without calling smart_global *)
+let set_strategy ~local l =
+  let glob_ref r =
+    match r with
+      | GlobRef.ConstRef sp -> EvalConstRef sp
+      | GlobRef.VarRef id -> EvalVarRef id
+      | _ -> user_err Pp.(str
+          "cannot set an inductive type or a constructor as transparent") in
+  let l = List.map (fun (lev,ql) -> (lev,List.map glob_ref ql)) l in
+  Redexpr.set_strategy local l
```

Comment:
This should probably be put somewhere else, behind a mli.

Reply:
The only difference between this and `vernac_set_strategy` is `match r with` vs `match smart_global r with`.  What do you think of moving this function to vernacentries (and exposing it in the .mli), and then making `vernac_set_strategy` first `map smart_global` and then call this function?

Reply:
Moved